### ISSUE TITLE
ChannelList won't get out of loading state after update to 5.6.0

### ISSUE THREAD

**Issue by 1987cr** [GetStream/stream-chat-react-native#1841]
**Describe the bug**
`ChannelList` will show the `LoadingIndicator` forever after updating library to v5.6.0 and react-native-quick-sqlite to v5.1.0, no other lines of codes have been changed. Some details:

- Seems to happen only the first time opening the app
- `LoadingIndicator` shows forever, no error on the console
- Sending the app to background and bringing it back to foreground seems to fix the issue
- After that the apps works correctly

Also, the following warning appears the first open after fresh install:

<img src="https://user-images.githubusercontent.com/4684912/205280191-b3c7cf5a-13b6-4263-8e2a-92675d4e1d37.png" width="300"  />

**Dev environment info (please complete/provide the following information):**

- stream-chat-react-native 5.6.0
- react-native 0.70.6
- react-native-quick-sqlite 5.1.0

**Comment by vishalnarkhede** [comment]
@1987cr do you see any error in console?

**Comment by 1987cr** [comment]
@vishalnarkhede nothing, downgraded to sqlite 4 just in case, and only a new deprecation warning when opening the app, will debug a little bit more.

**Comment by vishalnarkhede** [comment]
Thanks for picking this up. Please let me know if you find something, or can give us more info around it :)

**Comment by vishalnarkhede** [comment]
Could you check if you have installed 'stream-chat' dependency explicitly on app? If yes, better to drop it and let react native sdk install the necessary one from its dependency list

**Comment by 1987cr** [comment]
@vishalnarkhede we don't have the stream-chat dependency. After uninstalling and installing more times suddenly started working, we also had a broken patch package so that could be it (and sorry if that was the case haha 🙏🏻), but I'll do more digging tomorrow to see if I can reproduce the infinite loading state again, thanks!

**Comment by vishalnarkhede** [comment]
Thanks @1987cr  :) 

Closing this for now ... lets continue discussing though if you manage to reproduce it. I will reopen in that case :)

**Comment by 1987cr** [comment]
@vishalnarkhede we have been able to reproduce this on several devices, ios and android, simulator and real devices. Some details:

- Seems to happen only the first time
- `LoadingIndicator` shows forever, no error on the console
- Sending the app to background and bringing it back to foreground seems to fix the issue (maybe something related to `useIsOnline` reconnecting the websocket? just a wild guess)
- After that the apps works correctly

I'll try to debug stream channel list code now to see where's hanging exactly.

**Comment by vishalnarkhede** [comment]
Oh I see!

**Comment by 1987cr** [comment]
Updated prev comment with possible solution @vishalnarkhede

**Comment by vishalnarkhede** [comment]
We will need to set value for `syncStatus` for both the cases where `event.online` is true and is false.

```
if (event.online) {
        await this.syncAndExecutePendingTasks();
        this.syncStatus = true; <----------------------------------
        this.listeners.forEach((l) => l(true));
      } else {
        this.syncStatus = false; <----------------------------------
        this.listeners.forEach((l) => l(false));
      }
```

**Comment by vishalnarkhede** [comment]
@1987cr could you try `5.6.1-beta.1` and see if all seems fine?

**Comment by vishalnarkhede** [comment]
:tada: A fix for this issue is included in version 5.6.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/stream-chat-react-native-core/v/5.6.1)
- [GitHub release](https://github.com/GetStream/stream-chat-react-native/releases/tag/v5.6.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Comment by augustosamame** [comment]
We're suddenly, out of the blue, being hit by this issue. We're on version 5.9.0. It only happens on Android in real devices. Android and iOS simulators, and iOS real devices are unaffected. Any idea what you would need in order to debug this?
